### ISSUE TITLE
Preprocessing before Deseq2

### ISSUE THREAD

**Issue by cadl0590** [joey711/phyloseq#421]
Dear Joey,

Thank you for developing phyloseq. I have been using phyloseq and find it extremely useful. I was drawn to using phyloseq because of your ‘Waste not, Want not’ paper, in particular the use of deseq2 to normalise otu count data. 

I have a question in regards to pre-processing of otu count data before converting to a deseq2 dataset. I apologies in advance if this is covered in the supplementary material, however I am new to R and have probably just not understood some of the code.

In the simulation-cluster-accuracy-server supplement, under the heading ‘Naïve proportions’, there is mention about the need to preprocess the data (rm zeros, normalise, scale and centre). I wasn’t sure if this was done on the count data used for deseq2. Also under the heading ‘Perform normalisations’ there is some removal of low prevalence otus.

So basically, I wasn’t sure if you did do some of the above mentioned pre-processing to the otu count data before running deseq/2 or if you have any recommendations in regards to this issue.

Thanks in advance for your time and any help
All the best
Christina

**Comment by joey711** [comment]
Hi Christina,

Thanks for your positive feedback regarding phyloseq! 

Of the components of preprocessing that you mentioned, filtering low-abundance OTUS, normalize, scale, and center are all "handled" appropriately by the `DESeq2::DESeq` workflow when you test for differential abundance. For sample-wise clustering this is a little bit trickier to describe in general, but I have found that the regularized-log transformation is a good trade-off between speed and optimal variance stabilization. Furthermore, if sample clustering is your only aim and you're worried about fitting the model badly in a complex experimental design, then the 'naive proportions' or even a simple log/log-like transform of the proportions usually performs okay, or even really well, depending on the "noisiness" of your data. Most reasonable transformations will outperform rarefying during sample-clustering, because the latter ignores lots of available data. 

I think if you google `rlog` and `phyloseq` you might come across an example or two for the one that I like. Whether you need to re-scale, center, or filter OTUs will depend on your data and what you want to do with it. Most OTU count datasets that I've seen benefit from a bit of filtering.

Hope that helps. Feel free to post if you have further questions, but I'll close for now.

joey

**Comment by cadl0590** [comment]
Hi Joey,

Thanks for your reply, it is really helpful. I will have a think about my filtering approach.

For my project I am looking for differentially abundant OTUs in a fairly simple experimental design, which is testing the effect of diet on the oral microbiota of cats. I was also planning to use the variance stabilised counts for clustering (weighted unifrac) and I will try the regularised log transformation on these as well. Basically I am wanting to use the variance stabilised counts in place of wherever I would have previously used rarefied data. Hope that sounds sensible. 

Thanks again
All the best
Christina

**Comment by joey711** [comment]
A common misunderstanding is that the counts need to be transformed for differential abundance testing. This is not the case. The `DESeq2::DESeq` wrapper function is modeling both Poisson noise and biological variability based on a model that expects the original "raw" counts. However, this test is not relevant for whole-sample (whole microbiome) comparisons, but the underlying noise model is still appropriate for estimating the parameters that define an efficient variance stabilizing transformation function. 

Keep in mind that, despite the metaphorical love-affair with UniFrac/PCoA in the microbiome literature, whole-sample comparisons and PCA/PCoA are of limited utility in a well-defined hypothesis-derived experimental design -- where they serve mainly as quality assurance of the experiment and sample labels. For instance, you controlled diet because you already knew it had an effect; and even if you didn't know that a priori, you'd still need to be able to determine what microbes changed significantly between your design classes (if any), whether or not the UniFrac distance patterns told you they were different. 

Hope that helps clarify. Thanks again for the feedback.

**Comment by 56scientist** [comment]
Hi Joey,

Pardon me if it has already been answered,  the question is whether phyloseq can download sra data along with metadata, as it does for qiita.

**Comment by joey711** [comment]
If you don't mind, I'd rather answer this as a new issue, rather than as a comment at the end of this closed issue.
